data/scripts/agent.ts

```diff
@@ -0,0 +1,99 @@
+import fs from "fs";
+import path from "path";
+import cp from "child_process";
+import { VellumClient } from "vellum-ai";
+import { z } from "zod";
+import parseZodError from "package/utils/parseZodError";
+
+const zInput = z.object({
+  parameters: z.record(z.string()),
+  commands: z
+    .object({
+      exec: z.string(),
+      result: z.string(),
+    })
+    .array(),
+});
+
+const zOutput = z.object({
+  next: z.record(z.string()),
+  commands: z.string().array(),
+});
+
+const run = async () => {
+  const vellum = new VellumClient({
+    apiKey: process.env.VELLUM_API_KEY!,
+  });
+  const root = process.cwd();
+  const workspaceRoot = path.join(root, "workspace");
+  const nextInputLog = path.join(workspaceRoot, "next-inputs.json");
+  const killSwitch = path.join(workspaceRoot, "kill.log");
+  fs.writeFileSync(
+    nextInputLog,
+    JSON.stringify({ parameters: [], commands: [] })
+  );
+  const checkLoop = () => {
+    const killLoop = fs.existsSync(killSwitch);
+    if (killLoop) {
+      console.log("Kill switch found. Exiting...");
+      return false;
+    }
+    console.log(new Date().toISOString(), "Starting a new agent loop...");
+    return true;
+  };
+
+  while (checkLoop()) {
+    const input = zInput.parse(
+      JSON.parse(fs.readFileSync(nextInputLog).toString())
+    );
+    const response = await vellum.executeWorkflow({
```

Comment:
Gonna have as much logic live on vellum as possible. In future worlds, both the workflow runner logic and the natural language model should live on the box itself too. I'm gonna have one employee in Vellum's cloud and one employee in SamePage's cloud.